app/code/Magento/Wishlist/view/frontend/layout/wishlist_index_share.xml

```diff
@@ -9,7 +9,19 @@
     <update handle="customer_account"/>
     <body>
         <referenceContainer name="content">
-            <block class="Magento\Wishlist\Block\Customer\Sharing" name="wishlist.sharing" template="Magento_Wishlist::sharing.phtml" cacheable="false"/>
+            <block class="Magento\Wishlist\Block\Customer\Sharing" name="wishlist.sharing" template="Magento_Wishlist::sharing.phtml" cacheable="false">
+                <block class="Magento\Captcha\Block\Captcha" name="captcha" after="-" cacheable="false" ifconfig="customer/captcha/enable">
```

Comment:
With this layout update, we still have a soft dependency. So it can not be removed from `app/code/Magento/Wishlist/etc/module.xml`.

Reply:
Yeah I agree, while the dependency isn't explicit in module.xml and it will work with Captcha disabled, it's still a dependency

I was looking on how Magento_Customer does it, and it's actually the Magento_Captcha module that adds the block (and then it has a dependency on Magento_Customer)

I didn't had time to deep dive into this issue yet, but do you have any suggestions?

I could move all this logic to the Captcha module but feels weird having Magento_Wishlist as a dependency of Magento_Captcha

Another approach would be an intermediary module Magento_WishlistCaptcha that depends on both and enables Captcha on Wishlist sharing

What do you think?